app/controllers/tags_controller.rb

```diff
@@ -483,6 +488,9 @@ def prev_page_params
   def url_method(opts = {})
     if opts[:category_slug_path_with_id]
       "tag_category_#{action_name}_path"
+      # expect tag intersection if multiple tags are present
+    elsif opts[:tags] && opts[:tags].length > 1
+      "tag_intersection_path"
```

Comment:
How about checking for `params[:additional_tag_ids]` here? That way, we won't accidentally trigger the logic for genuine request with two `tags[]` parameters.